mac/build.sh

```diff
@@ -312,6 +115,30 @@ execBuildCommand() {
     fi
 }
 
+do_clean ( ) {
+  rm -rf "$KME4M_BASE_PATH/build"
+  rm -rf "$KM4MIM_BASE_PATH/build"
+  rm -rf "$KMTESTAPP_BASE_PATH/build"
+  rm -rf "$KEYMAN_ROOT/mac/setup/Install Keyman.app"
+
+  builder_heading "Cleaning pods folder (CocoaPods)"
+  rm -rf "$PODS_FOLDER"
+  builder_heading "Cleaning source (Carthage)"
+  rm -rf "$KEYMAN_MAC_BASE_PATH/Carthage"
```

Comment:
I personally find it odd to remove `configure` artifacts during the `clean` action as is done here.  That said, the _original_ version of the script's `clean`... only cleaned the Carthage output.

I'm making a point of this to ask:  should I drop these two lines from the `clean` action and keep the others?  Or should I keep them here to stick closer to _this_ script's original version, even if we don't clean `configure` outputs in other scripts?
- Keyman Engine for Web scripts never clean `configure` outputs.
- I'm fairly sure our iOS and Android platforms' scripts similarly don't clean `configure` outputs.
- Do our `windows`, `developer`, or `linux` platform build scripts ever do so?

Reply:
We're not 100% consistent on it, and probably can't be, because of the variety of what goes in `configure` steps. 

* `clean` on core (meson) removes the `build/<arch>` folder which is created in the `configure` step.
* `clean` on windows and developer, depends on the module type:
  * node, no, because that's is generally `/node_modules`
  * C++, meson toolchain, yes, removes `build/<arch>` folder which is created in the `configure` step
  * C++, msbuild toolchain, no, because configuration is shared across modules in `/resources/build/win`
  * Delphi, no, because configuration is shared across modules in `/resources/build/win`

I'm not too worried about it. The build scripts are pretty good at picking up the pieces after a `build.sh clean` and that's the main thing. 

Also, Carthage has been a problem in the past with instability, which may be why the `clean` action removed everything there too. Personally, I'd leave the clean of Carthage and Cocoapods in this script for now.

And of course, `git clean -fdx` is the hardcore clean tool to really deep clean the repo. Such a fresh new feeling.